lib/vagrant/environment.rb

```diff
@@ -564,6 +565,11 @@ def push(name)
           pushes: push_registry.keys
       end
 
+      # Validate the global push configuration and our local push configuration
+      machine = self.machine(self.machine_names.first, self.default_provider)
+      machine.action_raw(:config_validate, Vagrant::Action::Builtin::ConfigValidate)
+      config.validate(machine)
```

Comment:
Yeah here it is, we need to do something like this for the push root: https://github.com/mitchellh/vagrant/blob/master/plugins/kernel_v2/config/vm.rb#L671